scipy/interpolate/_rbfinterp.py

```diff
@@ -0,0 +1,538 @@
+        no minimum degree. Set this to -1 for no added polynomial.
+
+    Notes
+    -----
+    An RBF is a scalar valued function in N-dimensional space whose value at
+    :math:`x` can be expressed in terms of :math:`r=||x - c||`, where :math:`c`
+    is the center of the RBF.
+
+    An RBF interpolant for the vector of observations :math:`d`, which are made
+    at the locations :math:`y`, is a linear combination of RBFs centered at
+    :math:`y` plus a polynomial with a specified degree. The RBF interpolant is
+    written as
+
+    .. math::
+        f(x) = K(x, y) a + P(x) b
+
+    where :math:`K(x, y)` is a matrix of RBFs with centers at :math:`y`
+    evaluated at the interpolation points :math:`x`, and :math:`P(x)` is a
+    matrix of monomials, which span polynomials with the specified degree,
+    evaluated at :math:`x`. The coefficients :math:`a` and :math:`b` are the
+    solution to the linear equations
+
+    .. math::
+        (K(y, y) + \\lambda I) a + P(y) b = d
+
+    and
+
+    .. math::
+        P(y)^T a = 0,
+
+    where :math:`\\lambda` is a non-negative smoothing parameter that controls
+    how well we want to fit the observations. The observations are fit exactly
+    when the smoothing parameter is 0.
+
+    The above system is uniquely solvable if the following requirements are
+    met:
+
+        - :math:`P(y)` must have full column rank. When `degree` is 0,
+          :math:`P(y)` always has full column rank provided that there is at
+          least 1 data point. When `degree` is 1, :math:`P(y)` has full column
+          rank if there are N+1 data points and they are not collinear (N=2),
+          coplanar (N=3), etc.
+        - If `kernel` is 'mq', 'linear', 'tps', 'cubic', or 'quintic', then
+          `degree` must not be lower than the minimum value listed above.
+        - If `smoothing` is 0, then each data point location must be distinct.
+
+    When using an RBF that is not scale invariant ('mq', 'imq', 'iq', or 'ga'),
+    an appropriate shape parameter must be chosen (e.g., through cross
+    validation). Smaller values for the shape parameter correspond to wider
+    RBFs. The problem can become ill-conditioned or singular when the shape
+    parameter is too small.
+
+    See Also
+    --------
+    KNearestRBFInterpolator
+
+    References
+    ----------
+    .. [1] Fasshauer, G., 2007. Meshfree Approximation Methods with Matlab.
+        World Scientific Publishing Co.
+
+    .. [2] http://amadeus.math.iit.edu/~fass/603_ch3.pdf
+
+    .. [3] Wahba, G., 1990. Spline Models for Observational Data. SIAM.
+
+    .. [4] http://pages.stat.wisc.edu/~wahba/stat860public/lect/lect8/lect8.pdf
+
+    Examples
+    --------
+    Demonstrate interpolating scattered data to a grid in 2-D
+
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy.interpolate import RBFInterpolator
+    >>> np.random.seed(0)
+
+    >>> xobs = np.random.uniform(-1, 1, (100, 2))
```

Comment:
We don't write explicit seeds in examples (there is an automatic override which adds the seed).
```suggestion
    >>> rng = np.random.default_rng()

    >>> xobs = rng.uniform(-1, 1, (100, 2))
```

Reply:
Replaced with Halton sequence as suggested below. I think the resulting plot looks a bit nicer that way. BTW thank you for adding that to scipy!